stdlib.sh

```diff
@@ -302,7 +302,15 @@ find_up() {
 #
 # NOTE: the other ".envrc" is not checked by the security framework.
 source_env() {
-  local rcpath=${1/#\~/$HOME}
+  local rcpath
+
+  if [[ "$1" =~ ^[a-zA-Z]:.* ]]; then
+    # Go will pass native windows path when executed under Git Bash/Git for Windows
+    rcpath=$(type cygpath > /dev/null 2>&1 && cygpath -u "${1/#\~/$HOME}" 2> /dev/null)
```

Comment:
What does the `cygpath -u` command do?

If `type cygpath` returns false then the rcpath will be empty isn't it?

Reply:
https://cygwin.com/cygwin-ug-net/cygpath.html, converts local Windows path to corresponding  unix path. 
`cygpath` is available on cygwin like "bash" environment distributions for Windows. I don't want to assume it's existence, checked the most poppular https://gitforwindows.org/ & https://cmder.net/ full distribution. 

Yes, the `rcpath` path will be empty if `cygpath` doesn't exists, I could write out some information with `log_status` if that happens. If this shows to be an issue in the future, the idea is to replace `cygwin` path conversion with a similar construction like the current one for *nix systems 
https://github.com/direnv/direnv/blob/0c2afcc9fb9aec55c714844ce9760d5b43eb960d/stdlib.sh#L205-L220

Reply:
What do you think of always running it through cygpath, but only if it exists?

```
local rcpath=${1/#\~/$HOME}
if has cygpath ; then
  rcpath=$(cygpath -u "$rcpath")
fi
```

Reply:
`Command 'has' not found,` 
on ubuntu and msys ... needs probably be something like this 
```
if type cygpath > /dev/null 2>&1; then 
  rcpath=$(cygpath -u "$rcpath")
fi
```
Otherwise I like the construction, i need to run some scenarios and update this PR if it works as expected.

Reply:
[has](https://github.com/direnv/direnv/blob/0c2afcc9fb9aec55c714844ce9760d5b43eb960d/stdlib.sh#L166-L179) is part of the stdlib :)